oak_functions/loader/src/server.rs

```diff
@@ -153,15 +153,26 @@ pub trait ExtensionFactory {
 /// `Uwabi` extension called by listening to a channel.
 pub enum BoxedExtension {
     Native(Box<dyn OakApiNativeExtension + Send + Sync>),
-    Uwabi(Box<dyn UwabiExtension + Send + Sync>),
+    Uwabi(BoxedUwabiExtension),
 }
 
+pub type BoxedUwabiExtension = Box<dyn UwabiExtension + Send + Sync>;
+
 pub type BoxedExtensionFactory = Box<dyn ExtensionFactory + Send + Sync>;
 
 /// Trait for implementing an extension which relies on UWABI.
 pub trait UwabiExtension {
     /// Get the channel handle to address this extension.
     fn get_channel_handle(&self) -> ChannelHandle;
+
+    /// Get the endpoint.
+    fn get_endpoint(&mut self) -> Option<&mut Endpoint>;
```

Comment:
Why would anything other than the extension itself need access to the endpoint? It seems to break encapsulation.

Reply:
True, that is not very pretty. I need the access to the endpoint only for tests right now.

It may be a much better idea to add `channel_read` and `channel_write` to the `UwabiExension` trait to be able to read/write to an endpoint. WDYT?

Reply:
I added a TODO to remove `get_endpoint`, because this requires some more design decisions #2508 .